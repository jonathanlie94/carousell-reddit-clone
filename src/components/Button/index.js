import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { theme } from 'ui';

const StyledButton = styled.button`
  height: auto;
  text-align: center;
  font-family: Overwatch;
  font-size: ${theme.fontSizes.regularL};
  outline-width: 0;
  /* The paddings here are different from right and left, because we are using a
    a special font which makes the formatting unequal between sides
  */
  padding-left: ${theme.margins.small};
  padding-right: ${theme.margins.regular};
  color: ${props => props.color};
  border-color: transparent;
  border-radius: 4px;
  transform: skewX(-.25rad);
  background-color: ${props => props.backgroundColor};
  & > span {
    transform: skewX(.25rad);
    display: block;
  }
  &:focus:not([disabled]),
  :hover:not([disabled]) {
    opacity: .625;
  }
  &:disabled {
    opacity: .25;
  }
  transition: .1s ease;
`;

class Button extends Component {
  render() {
    return (
      <StyledButton
        className={this.props.className}
        color={this.props.color}
        disabled={this.props.disabled}
        backgroundColor={this.props.backgroundColor}
        onClick={this.props.disabled ? null : this.props.onClick || null}
      >
        <span>
          {this.props.children}
        </span>
      </StyledButton>
    );
  }
}

Button.propTypes = {
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  color: theme.colors.white,
  backgroundColor: theme.colors.blue,
};

export default Button;
