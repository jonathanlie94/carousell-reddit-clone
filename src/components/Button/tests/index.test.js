import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import Button from '../';

describe('<Button />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <Button>
        <div className="unique-class-name" />
      </Button>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });

  it('should render a button', () => {
    const button = shallow(<Button />).dive();
    expect(button.type()).toEqual('button');
  });

  it('should match previous snapshot', () => {
    const button = shallow(<Button />).dive();
    const tree = toJSON(button);
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it('calls onClick props when clicked', () => {
    const onClickSpy = jest.fn();
    const renderedComponent = mount(<Button onClick={onClickSpy} />);
    renderedComponent.find('button').simulate('click');
    expect(onClickSpy).toHaveBeenCalled();
  });
});
