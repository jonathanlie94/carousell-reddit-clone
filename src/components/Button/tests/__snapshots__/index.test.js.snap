// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match previous snapshot 1`] = `
.kaRKrY {
  height: auto;
  text-align: center;
  font-family: Overwatch;
  font-size: 1.25rem;
  outline-width: 0;
  padding-left: 0.75rem;
  padding-right: 1rem;
  color: #FFF;
  border-color: transparent;
  border-radius: 4px;
  -webkit-transform: skewX(-.25rad);
  -ms-transform: skewX(-.25rad);
  transform: skewX(-.25rad);
  background-color: #00C3FF;
  -webkit-transition: .1s ease;
  transition: .1s ease;
}

.kaRKrY > span {
  -webkit-transform: skewX(.25rad);
  -ms-transform: skewX(.25rad);
  transform: skewX(.25rad);
  display: block;
}

.kaRKrY:focus:not([disabled]),
.kaRKrY:hover:not([disabled]) {
  opacity: .625;
}

.kaRKrY:disabled {
  opacity: .25;
}

<button
  className="sc-bdVaJa kaRKrY"
  color="#FFF"
  onClick={null}
>
  <span />
</button>
`;
